### PR TITLE
feat: add support for Powershell Core, PhotonOS and pre-check

### LINKED ISSUE #27: Add support for PowerShell Core and VMware PhotonOS
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

Currently, the module only works on Windows with Windows Powershell 5.1. 

Add support for PowerShell Core and VMware PhotonOS/.

### Use Case(s)

The below modules need to be updated - 

- `Test-Connection` should use `-ComputerName` instead of `-server`.
- `Confirm-ESXiCertificateInstalled` needs a rework to verify certificates. 

```powershell
PS /root> Test-Connection -server sfo-vcf01.sfo.rainpole.io
Test-Connection: A parameter cannot be found that matches parameter name 'server'.
PS /root>

PS /root/.local/share/powershell/Modules> Confirm-ESXiCertificateInstalled -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re123! -esxiFqdn sfo01-w02-esx04.sfo.rainpole.io -signedCertificate /opt/Certificates/sfo01-w02-esx04.sfo.rainpole.io.cer
 Error at Script Line 275
 Relevant Command: $crt.Import($signedCertificate)
 Error Message: Exception calling "Import" with "1" argument(s): "X509Certificate is immutable on this platform. Use the equivalent constructor instead."
PS /root/.local/share/powershell/Modules>
```

### Potential Configuration

NA

### References

_No response_

### LINKED ISSUE #30: Add support for an ESXi certificate management pre-check
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

Create a function supporting all pre-checks to be run before performing certificate replacement. 

### Use Case(s)

NA

### Potential Configuration

NA

### References

_No response_

### PR DESCRIPTION
**Pull request by bhumitra**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-certificate-management/blob/main/CONTRIBUTING_DCO.md) for making a pull request.

**Summary of Pull Request**

- Add support for PowerShell Core and VMware PhotonOS.
- Add support for a ESXi certificate management pre-check.
- Enhanced `Get-vSANHealthSummary` by improving log messages and adding checks for vSAN services.
- Updated the case for functions from `ESXi...` to `Esxi`.

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Closes #27 
Closes #30

**Test and Documentation Coverage**

- [x] Tests have been completed (for bug fixes / features).
- [x] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
 
+# Release History
+
+## v1.1.0 (Not Released)
+
+> Release Date: Unreleased
+
+Enhancement:
+
+- Add support for PowerShell Core and VMware PhotonOS
```
Comment (tenthirtyam) [inline CHANGELOG.md:11]:
```suggestion
- Add support for PowerShell Core and VMware PhotonOS.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+- Add support for a pre-check
+- Enhanced Get-vSANHealthSummary by improving log messages and adding check for vSAN services
```
Comment (tenthirtyam) [inline CHANGELOG.md:13]:
```suggestion
- Enhanced `Get-vSANHealthSummary` by improving log messages and adding a check for vSAN services.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+# Release History
+
+## v1.1.0 (Not Released)
+
+> Release Date: Unreleased
+
+Enhancement:
+
+- Add support for PowerShell Core and VMware PhotonOS
+- Add support for a pre-check
```
Comment (tenthirtyam) [inline CHANGELOG.md:12]:
```suggestion
- Add support for an ESXi certificate management pre-check.
```

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
an update is needed for PowerShell Core and VMware PhotonOS in `docs/index.md`. 

Take a look at the reporting module's version of this file to reproduce this.

File: CHANGELOG.md (hunk 1 above)
Comment (tenthirtyam) [reply CHANGELOG.md]: Addressed in 82ea751197a80befea01e212a77aabc86ce15719.

File: CHANGELOG.md (hunk 3 above)
Comment (tenthirtyam) [reply CHANGELOG.md]: Addressed in 82ea751197a80befea01e212a77aabc86ce15719.

File: CHANGELOG.md (hunk 2 above)
Comment (tenthirtyam) [reply CHANGELOG.md]: Addressed in 82ea751197a80befea01e212a77aabc86ce15719.

File: VMware.CloudFoundation.CertificateManagement.psm1 [hunk 4]
```diff
@@ -1276,7 +1276,7 @@ Function Install-EsxiCertificate {
                     $esxCertificatePem = Get-Content $crtPath -Raw
                     Set-VIMachineCertificate -PemCertificate $esxCertificatePem -VMHost $esxiFqdn -ErrorAction Stop -Confirm:$false
                     $replacedHosts.Add($esxiFqdn)
-                    Set-EsxiConnectionState -esxiFqdn $esxiFqdn -state "Disconnected" -timeout $timeout
+                    #Set-EsxiConnectionState -esxiFqdn $esxiFqdn -state "Disconnected" -timeout $timeout
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.CertificateManagement.psm1]: @bhumitra - is there a reason that this was commented out here?

Comment (bhumitra) [reply VMware.CloudFoundation.CertificateManagement.psm1]: It should be uncommented. This is also outdated now. Closing it.

**Review by tenthirtyam — APPROVED** [review]
LGTM!

**Pull request merged**